src/utilities/fastify/handler/setErrorHandler.ts

```diff
@@ -0,0 +1,20 @@
+import type { FastifyInstance } from 'fastify';
+
+import { HTTP_STATUS_CODES } from '../../http.js';
+
+export default function setErrorHandler(fastify: FastifyInstance): void {
+  const internalServerError = 'Internal server error';
+  fastify.setErrorHandler(async (error, _request, reply) => {
+    if (
+      error.statusCode !== undefined &&
+      error.statusCode < HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR
+    ) {
+      fastify.log.info({ err: error }, 'Client error');
+      await reply.send(error);
+      return;
+    }
+
+    fastify.log.error({ err: error }, internalServerError);
```

Comment:
You should always use `request.log` where possible as it contains important context (e.g., request id), which `fastify.log` lacks (because it's agnostic of any particular request).